### PR TITLE
FEATURE: Store reply-to in draft

### PR DESCRIPTION
**Pull request by markvanlan**
@CvX I know you'll have the async in the test... but the chat float component waits on a promise to check the channels so... idk how else to do this shiz.

### CODE REVIEW

**Comment by CvX** [comment]
UX question: should it maybe save reply-to info in a message draft instead of just clearing it?

**Comment by markvanlan** [comment]
Yeah, that's a better idea isn't it

**Comment by pmusaraj** [comment]
I ran into something adjacent to this just now: when you hit arrow up to edit your last message but then hit Esc to cancel it, it cancels the reply state, but doesn't clear the input. It feels like it should?

**Comment by markvanlan** [comment]
Yeah I can clear the input there!

File: assets/javascripts/discourse/components/chat-live-pane.js [hunk 1]
```diff
@@ -908,14 +911,28 @@ export default Component.extend({
 
   @action
   _setDraftForChannel(draft) {
+    if (draft?.replyToMsg) {
+      draft.replyToMsg = {
+        id: draft.replyToMsg.id,
+        excerpt: draft.replyToMsg.excerpt,
+        user: draft.replyToMsg.user,
+      };
```
Comment (markvanlan) [inline assets/javascripts/discourse/components/chat-live-pane.js:919]: Here we don't care to store ALL the chat message data, just what we need to render the reply-to.

**Review by CvX — APPROVED** [review]

**Pull request merged**